File: templates/python/fr/text/code/moveToWall.py
# Fichier: MoveToWall.py
# ------------------------------
# Utilise une boucle «while» vers avancer Karel jusqu'à ce qu'elle avancer
# un mur. Fonctionne sur n'importe quel monde de taille.
from karel.stanfordkarel import *

# le programme commence par main
def main():
   # appeler la fonction avancer au mur
   avancer_au_mur()

# c'est une fonction très utile
def avancer_au_mur():
   # répéter le corps tant que la condition est maintenue
   while l'avant_est_clair():
      avancer()

File: templates/java/fr/text/code/moveToWall.java
/*
 * Fichier: AvancerAuMur.java
 * --------------------------
 * Utilise une boucle "while" de avancer Karel jusqu'
 * à ce qu'elle heurte un mur. Fonctionne sur n'impor
 * te quel monde de taille.
 */
import stanford.karel.*;
public class AvancerAuMur extends Karel {
   public void run() {
      avancerAuMur();
   }

   /* c'est une méthode très utile */
   private void avancerAuMur() {
      while(devantDegage()) {
         avancer();
      }
   } 
}